File: iota/go/iota.go
package main

import "fmt"

type Iota func()int

func NewIota(start int) Iota{
	return func() int {
		start += 1
		return start - 1
	}
}

func main() {
	enum := NewIota(0)
	for i := 0; i < 10; i += 1 {
		fmt.Println("it: ", enum())
	}
}

File: code/iota/js/iota.js
function NewIota(start){
	start -= 1;
	return function(){
		start += 1;
		return start;
	}
}

iota = NewIota(0)
iota() // 0
iota() // 1
iota() // 2

enum = NewIota(0)
var itemTyp = {
	Error : enum(),
	Bool : enum(),
	Complex : enum(),
	EOF : enum()
}
